File: Nurtibudget GUI/src/nutrient_ranges.py
NUTRIENT_RANGES = {
    # format: (RDI, Upper Limit (UL))

    # Vitamins
    "vitamin_a": (900, 3000),
    "vitamin_b1": (1.2, 50),
    "vitamin_b2": (1.3, 50),
    "vitamin_b3": (16, 35),
    "vitamin_b5": (5, 1000),
    "vitamin_b6": (1.3, 100),
    "vitamin_b7": (30, 1000),
    "vitamin_b9": (400, 1000),
    "vitamin_b12": (2.4, 1000),
    "vitamin_c": (90, 2000),
    "vitamin_d": (20, 100),
    "vitamin_e": (15, 1000),
    "vitamin_k": (120, 1000),

    # Minerals
    "calcium": (1000, 2500),
    "phosphorus": (700, 4000),
    "magnesium": (400, 350),  # Note: UL applies to supplements, use carefully
    "sodium": (1500, 2300),
    "potassium": (4700, 5000),
    "chloride": (2300, 3600),
    "sulfur": (1000, 2000),
    "iron": (18, 45),
    "zinc": (11, 40),
    "copper": (0.9, 10),
    "manganese": (2.3, 11),
    "iodine": (150, 1100),
    "selenium": (55, 400),
    "molybdenum": (45, 2000),
    "chromium": (35, 1000),
    "fluoride": (4, 10),
    "cobalt": (5, 100),

    # Macronutrient bonuses
    "fiber": (25, 70),
    "protein": (50, 200),
}

File: Nurtibudget GUI/src/pages/nutrientRanges.java
package pages;


import java.util.HashMap;
import java.util.Map;

public class nutrientRanges {

    public static final Map<String, double[]> NUTRIENT_RANGES = new HashMap<>();

    static {
        // Format: {RDI, UL}

        // Vitamins
        NUTRIENT_RANGES.put("vitaminA", new double[]{900, 3000});
        NUTRIENT_RANGES.put("vitaminB1", new double[]{1.2, 50});
        NUTRIENT_RANGES.put("vitaminB2", new double[]{1.3, 50});
        NUTRIENT_RANGES.put("vitaminB3", new double[]{16, 35});
        NUTRIENT_RANGES.put("vitaminB5", new double[]{5, 1000});
        NUTRIENT_RANGES.put("vitaminB6", new double[]{1.3, 100});
        NUTRIENT_RANGES.put("vitaminB7", new double[]{30, 1000});
        NUTRIENT_RANGES.put("vitaminB9", new double[]{400, 1000});
        NUTRIENT_RANGES.put("vitaminB12", new double[]{2.4, 1000});
        NUTRIENT_RANGES.put("vitaminC", new double[]{90, 2000});
        NUTRIENT_RANGES.put("vitaminD", new double[]{20, 100});
        NUTRIENT_RANGES.put("vitaminE", new double[]{15, 1000});
        NUTRIENT_RANGES.put("vitaminK", new double[]{120, 1000});

        // Minerals
        NUTRIENT_RANGES.put("calcium", new double[]{1000, 2500});
        NUTRIENT_RANGES.put("phosphorus", new double[]{700, 4000});
        NUTRIENT_RANGES.put("magnesium", new double[]{400, 350}); // careful: UL applies to supplements
        NUTRIENT_RANGES.put("sodium", new double[]{1500, 2300});
        NUTRIENT_RANGES.put("potassium", new double[]{4700, 5000});
        NUTRIENT_RANGES.put("chloride", new double[]{2300, 3600});
        NUTRIENT_RANGES.put("sulfur", new double[]{1000, 2000});
        NUTRIENT_RANGES.put("iron", new double[]{18, 45});
        NUTRIENT_RANGES.put("zinc", new double[]{11, 40});
        NUTRIENT_RANGES.put("copper", new double[]{0.9, 10});
        NUTRIENT_RANGES.put("manganese", new double[]{2.3, 11});
        NUTRIENT_RANGES.put("iodine", new double[]{150, 1100});
        NUTRIENT_RANGES.put("selenium", new double[]{55, 400});
        NUTRIENT_RANGES.put("molybdenum", new double[]{45, 2000});
        NUTRIENT_RANGES.put("chromium", new double[]{35, 1000});
        NUTRIENT_RANGES.put("fluoride", new double[]{4, 10});
        NUTRIENT_RANGES.put("cobalt", new double[]{5, 100});

        // Macronutrients
        NUTRIENT_RANGES.put("fiber", new double[]{25, 70});
        NUTRIENT_RANGES.put("protein", new double[]{50, 200});
    }
}
